### DISCUSSION TITLE
Packet compression

### DISCUSSION THREAD

**Discussion by Therzok** [NebulaModTeam/nebula#49] [Ideas]
I was looking up a few blog posts, might be useful reads:

https://gafferongames.com/post/snapshot_compression/
https://straypixels.net/better-networktransform/

**Comment by Flapperkewiet** [comment]
Hmmm, interesting articles, but they assume control over the data structures to send. 

Those tricks might be nice to use for our own packets, but the data the game generates (from `Export()` functions) are `byte[]`.

We could implement this if we export the data ourselfs, which would be a whole lot of work, but I would definitely suggest looking into it.

**Reply by Therzok** [reply]
Yeah, using the `Export` mechanism might be forward compatible by leveraging the game dev's work done on the game save logic.

Maybe having a fallback `Export` method in case the game save version number changes between versions to provide forward-compatibility, and manually optimizing where possible?
